ElectronClient/app/gui/NoteText.jsx

```diff
@@ -897,6 +884,41 @@ class NoteTextComponent extends React.Component {
 		});
 	}
 
+	commandSavePdf() {
+		const path = bridge().showSaveDialog({
+			filters: [{ name: _('PDF File'), extensions: ['pdf']}],
+			defaultPath: safeFilename(this.state.note.title),
+		});
+
+		if (path) {
+			// temporarily change the html in the webview to show the note title
+			const newHtml = this.insertHtmlHeading_(this.lastSetHtml_, this.state.note.title);
+			this.webview_.send('setHtml', newHtml);
+
+			this.webview_.printToPDF({}, (error, data) => {
+				if (error) {
+					bridge().showErrorMessageBox(error.message);
+				} else {
+					shim.fsDriver().writeFile(path, data, 'buffer');
+				}
+
+				// refresh the webview contents, undoing our temporary change
+				this.lastSetHtml_ = '';
+				this.render()
+			});
+		}
+	}
+
+	insertHtmlHeading_(s, heading) {
+		const tag = 'h2';
+		let splitStyle = s.split('</style>');
```

Comment:
Hmm... what is later on a style block is added at the bottom of a note - wouldn't that mean that the title would end up there too? Is it not possible to insert the title based on the <body>tag? Or maybe a hidden placeholder could be added to the generated HTML for example as a comment - `<!-- TITLE_PLACEHOLDER -->` then this placeholder would be replaced here.

Reply:
Not sure what you mean by "insert the title based on the tag"? the code as written inserts the heading after the first </style>, so it would still work as expected if there are other </style> blocks.

I'm now using a comment placeholder now which is less fragile in the long term.